packages/hardhat-core/test/internal/hardhat-network/jsonrpc/client.ts

```diff
@@ -23,402 +25,450 @@ type FakeProvider = Pick<HttpProvider, "url" | "sendBatch"> & {
 };
 
 describe("JsonRpcClient", () => {
-  FORKED_PROVIDERS.forEach(({ rpcProvider, jsonRpcUrl }) => {
-    describe(`Using ${rpcProvider}`, () => {
-      let client: JsonRpcClient;
+  describe("Using fake providers", function () {
+    describe("Caching", () => {
+      const response1 =
+        "0x00000000000000000000000000000000000000000067bafa8fb7228f04ffa792";
+      const response2 =
+        "0x00000000000000000000000000000000000000000067bafa8fb7228f04ffa793";
 
-      beforeEach(() => {
-        client = JsonRpcClient.forUrl(jsonRpcUrl);
+      let fakeProvider: FakeProvider;
+      let clientWithFakeProvider: JsonRpcClient;
+
+      function getStorageAt(blockNumber: number) {
+        return clientWithFakeProvider.getStorageAt(
+          DAI_ADDRESS,
+          DAI_TOTAL_SUPPLY_STORAGE_POSITION,
+          new BN(blockNumber)
+        );
+      }
+
+      beforeEach(async function () {
+        fakeProvider = {
+          request: sinon.stub().resolves(response1),
+          url: "fake",
+          sendBatch: () => Promise.resolve([]),
+        };
       });
 
-      describe("Basic tests", () => {
-        it("can be constructed", () => {
-          assert.instanceOf(client, JsonRpcClient);
-        });
+      it("Doesn't cache things for blocks that can be reorg'd out", async () => {
+        clientWithFakeProvider = new JsonRpcClient(
+          fakeProvider as any,
+          1,
+          123,
+          3
+        );
 
-        it("can actually fetch real json-rpc", async () => {
-          const result = await client.getLatestBlockNumber();
-          const minBlockNumber = 10494745; // mainnet block number at 20.07.20
-          assert.isAtLeast(result.toNumber(), minBlockNumber);
-        });
+        const value = await getStorageAt(121);
+        await getStorageAt(121);
```

Comment:
Is it worth it to add an assertion that this result is equal to `value`?

Reply:
makse sense